.github/workflows/create-release-pr.yaml

```diff
@@ -0,0 +1,328 @@
+      - uses: actions/checkout@v4
+      #----------------------------------------------
+      #       Setup python
+      #----------------------------------------------
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+      #----------------------------------------------
+      #       Install poetry
+      #----------------------------------------------
+      - name: Install poetry
+        run: pipx install poetry
+        id: setup-poetry
+      #----------------------------------------------
+      #       Get the latest version of aries-cloudagent from pypi
+      #----------------------------------------------
+      - name: Get latest aries-cloudagent version
+        id: current_available_version
+        run: |
+          remote_version=$(pip index versions aries-cloudagent)
+          version=$(grep -oP '(?<=Available versions: ).*?(?=,)' <<< "$remote_version")
+          echo current_available_version=$version >> $GITHUB_OUTPUT
+          echo "Remote version = $version"
+      #----------------------------------------------
+      #       Check the latest version from plugins_global lock file
+      #----------------------------------------------
+      - name: Get global aries-cloudagent version from plugins repo
+        id: current_global_version
+        run: |
+          cd plugin_globals
+          lock_version=$(grep -A1 'name = "aries-cloudagent"' poetry.lock | grep -v 'name = "aries-cloudagent"')
+          version=$(grep -oP '(?<=").*?(?=")' <<< "$lock_version")
+          echo current_global_version=$version >> $GITHUB_OUTPUT
+          echo "Global version = $version"
+      #----------------------------------------------
+      #       Check if aries-cloudagent upgrade available
+      #       If the global version is greater than or equal to the remote version, exit
+      #       If re-release is requested, skip this step
+      #----------------------------------------------
+      - name: Check if aries-cloudagent upgrade available
+        run: |
+          current_available_version="${{steps.current_available_version.outputs.current_available_version}}"
+          echo "Remote version = $current_available_version"
+          current_global_version="${{steps.current_global_version.outputs.current_global_version}}"
+          echo "Global version = $current_global_version"
+          sem_version () { 
+            echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; 
+          }
+          echo "Input re-release = ${{ inputs.re_release }}"
+          if [ ${{ inputs.re_release }} == "true" ]; then
+            echo "Re-Release requested"
+          else
+            if [ $(sem_version $current_global_version) -ge $(sem_version $current_available_version) ]; then
+              echo "Version of aries-cloudagent is up to date"
+              exit 1
+            fi
+              echo "Detected aries-cloudagent upgrade available..."
+          fi
```

Comment:
Would it be better to use conditional logic on the step rather than checking `inputs.re_release` inside the script?